brave/brave-core
# Adds support for direct and recurring donations

Resolves https://github.com/brave/brave-browser/issues/925
Resolves https://github.com/brave/brave-browser/issues/1550

## Submitter Checklist:

- Verified that these changes build without errors on
- Verified that these changes pass automated tests (`npm test brave_unit_tests && npm test brave_browser_tests`) on

## Test Plan:
(for developers only)
Uncomment this line https://github.com/brave-intl/bat-native-ledger/blob/master/src/bat_client.cc#L183 to trigger contribution after 5min

1.
- clean profile
- enable rewards
- claim grant
- visit some sites (wait 10s on them)
- wait for contribution to happen

2.
- clean profile
- enable rewards
- claim grant
- visit some sites and add them to recurring donation via panel and site banner
- wait for contribution to happen

3.
- clean profile
- enable rewards
- claim grant
- visit some sites (wait 10s on them)
- visit some sites and add them to recurring donation via panel and site banner
- wait for both contribution to happen

4.
- clean profile
- enable rewards
- claim grant
- visit site
- click on panel, donate now and pick amount
- make sure that direct donation happens

## Notes

1. Currently you can't do recurring donation and ac at the same time. You will get error like this `Error: the signed and unsigned transactions differed`. This will be fixed on Monday (15/10) on the server


## Reviewer Checklist:
